src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -345,5 +369,46 @@ internal bool IsNetStandardFeatureFlagEnabled()
 
             return value == "1" || value == "true";
         }
+
+        internal IDictionary<int, bool> ParseHttpCodesToDictionary(string httpStatusErrorCodes)
+        {
+            string[] configurationsArray = string.Concat(httpStatusErrorCodes.Where(c => !char.IsWhiteSpace(c))).Split(',');
```

Comment:
Can we replace this with `Split` removing empty items + `Trim` inside the `foreach`?

Reply:
In this case I also need to take into consideration when a value like "401 - 404, 419" is sent so it can be parsed as "401-404,419", how do you recommend I should approach it? I was maybe thinking:
```
string[] configurationsArray = httpStatusErrorCodes.Split();
configurationsArray = string.Join(string.Empty, configurationsArray).Split(',');
```

Reply:
That works but I guess we can allocate less than two arrays and a string.

Maybe worth checking a simple `string.Replace(" ", string.Empty).Split(',')` That would be 1 string and 1 array. 
`string.Replace` is an [internal call](https://referencesource.microsoft.com/#mscorlib/system/string.cs,2877), so I don't know the hidden cost of that operation.

Also TIL, `string.Join` uses [`fixed`](https://referencesource.microsoft.com/#mscorlib/system/string.cs,253)

Reply:
I wouldn't worry too much about allocations since this happens only once, during startup. Without optimizing, I would start with something like this. I'm calling `401-404` a "range" composed of two "limits" (`start` and `end`).

```csharp
string setting = "401 - 404 , 450";
bool[] flags = new bool[600];

string[] ranges = setting.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries);

foreach(string range in ranges)
{
    string[] limits = range.Split(new[] {'-'}, StringSplitOptions.RemoveEmptyEntries);
    
    if(limits.Length > 2)
    {
        // TODO: handle error
    }
    
    int start = int.Parse(limits[0]); // TODO: handle error
    
    if(limits.Length == 1)
    {
        // single value, e.g. 404
        flags[start] = true;
    }
    else
    {
        // range, e.g. 400-499
        int end = int.Parse(limits[1]); // TODO: handle error

        for (int x = start; x <= end; x++)
        {
            flags[x] = true;
        }
    }
}
```
Thoughts?

Reply:
👍 

I guess the TODOs can be handled with the help of `int.TryParse`

Reply:
Definitely. Please, consider this snippet as pseudo-code 😉 
No guarantee that it works or even compiles, hehe.